### ISSUE TITLE
[FEATURE] Add a way to stop the live feed event loop

### ISSUE THREAD

**Issue by BeatsuDev** [BeatsuDev/tibber.py#9]
## Is your feature request related to a problem? Please describe.
Once you've started the home live feed loop with `home.start_live_feed()`, there is no way of stopping it. This would be useful for unit tests and could be handy in other places too.

### Conditional-based approach
Pass a condition or a function that is evaluated every time a message is received. If it is evaluated to True, the loop ends.
```python
# Exit after 10 seconds
start_time = time.perf_counter()
# Note that data will always be the first parameter of the condition function.
# This way, the live feed can exit based on data conditions
def condition(data):
    return time.perf_counter() - start_time > 10
home.start_live_feed(exit_condition = condition)
```
```python
# Exit when the accumulated cost goes above 10
home.start_live_feed(exit_condition = lambda data: data.accumulated_cost > 10)
```

### Time-based approach
Exit after x seconds.
```python
# Exit after 10 seconds
home.start_live_feed(exit_after = 10)
```

### Iteration-based approach
Exit after x amount of iterations.
```python
# Exits after data has been received 10 times
home.start_live_feed(exit_after = 10)
```

## Describe the solution you'd like
The iteration-based approach is more unpredictable because the time at which data is received is not always known or consistent. Also, it could be possible to exit after x amount of iterations by using the conditional-based approach. 

The time-based approach does not guarantee that data will be received within the timeframe. 

I would therefore suggest adding at least the conditional-based approach and the time-based approach in case a user
would for some reason want to exit after a certain amount of time (which is a bit clunky to do using the conditional-based approach).

**Comment by JoshuaDodds** [comment]
Hi @BeatsuDev,

I would just like to add that the livefeed event loop also is not very resilient to outages of the Tibber API or other network issues.  In other words once the feed is interrupted or fail for any reason, it does not restart.  For this reason when one is using this live feed as a long running service it might be better to use this feature you suggest and have the live feed start and poll for a number of secs or iterations and then exit.  If we then put these steps in a loop, we have something that will self recover.  

Is there any planning for when this feature might be added ?

**Comment by BeatsuDev** [comment]
> 

Sorry for the late response! I'll definitely look into it as soon as I have time. You have a great point. There's no plan for now, but I'll change some things about it as soon as I can.

**Comment by BeatsuDev** [comment]
> Hi @BeatsuDev,
> 
> I would just like to add that the livefeed event loop also is not very resilient to outages of the Tibber API or other network issues. In other words once the feed is interrupted or fail for any reason, it does not restart. For this reason when one is using this live feed as a long running service it might be better to use this feature you suggest and have the live feed start and poll for a number of secs or iterations and then exit. If we then put these steps in a loop, we have something that will self recover.
> 
> Is there any planning for when this feature might be added ?

Added in 6efb414bd74e7ff533bf3ac28c911c29ff1ac1c2!

**Comment by BeatsuDev** [comment]
As for the issue, I've made this change: https://github.com/BeatsuDev/tibber.py/commit/9f7ac25547ac39d7d178e099dab4602463d74c11, implementing the conditional-based approach described in the issue description. I'll see how it works and run some extra tests tomorrow. If everything works fine, I'll consider uploading both fixes to pypi in version 0.3.0 or 0.2.1 (not quite sure yet) and closing this issue for now.

**Comment by BeatsuDev** [comment]
Added this in version 0.2.1 😄
